holidays/countries/new_zealand.py

```diff
@@ -179,96 +180,138 @@ def _populate(self, year):
-            self[
-                self._get_nearest_monday(date(year, FEB, 1))
-            ] = "Nelson Anniversary Day"
-
-        elif self.subdiv in {"CAN", "Canterbury"}:
-            self[
-                _get_nth_weekday_of_month(1, TUE, NOV, year) + td(days=+10)
-            ] = "Canterbury Anniversary Day"
-
-        elif self.subdiv in {"South Canterbury", "STC"}:
-            self[
-                _get_nth_weekday_of_month(4, MON, SEP, year)
-            ] = "South Canterbury Anniversary Day"
-
-        elif self.subdiv in {"WTC", "West Coast", "WTL", "Westland"}:
-            name = "West Coast Anniversary Day"
-            # Observance varies?!?!
-            if year == 2005:  # special case?!?!
-                self[date(year, DEC, 5)] = name
-            else:
-                self[self._get_nearest_monday(date(year, DEC, 1))] = name
-
-        elif self.subdiv in {"OTA", "Otago"}:
-            # there is no easily determined single day of local observance?!?!
-            dt = self._get_nearest_monday(date(year, MAR, 23))
-            if dt == easter_date + td(days=+1):  # Avoid Easter Monday
-                dt += td(days=+1)
-            self[dt] = "Otago Anniversary Day"
-
-        elif self.subdiv in {"STL", "Southland"}:
-            name = "Southland Anniversary Day"
-            if year >= 2012:
-                self[easter_date + td(days=+2)] = name
-            else:
-                self[self._get_nearest_monday(date(year, JAN, 17))] = name
+        dec_26 = self._add_christmas_day_two("Boxing Day")
+        self._add_observed(dec_25)
+        self._add_observed(dec_26)
+
+        if self.subdiv == "Auckland":
+            self._add_subdiv_auk_holidays()
+        elif self.subdiv == "Canterbury":
+            self._add_subdiv_can_holidays()
+        elif self.subdiv == "Chatham Islands":
+            self._add_subdiv_cit_holidays()
+        elif self.subdiv == "Hawke's Bay":
+            self._add_subdiv_hkb_holidays()
+        elif self.subdiv == "Marlborough":
+            self._add_subdiv_mbh_holidays()
+        elif self.subdiv == "Nelson":
+            self._add_subdiv_nsn_holidays()
+        elif self.subdiv == "Northland":
+            self._add_subdiv_ntl_holidays()
+        elif self.subdiv == "Otago":
+            self._add_subdiv_ota_holidays()
+        elif self.subdiv in {"New Plymouth", "Taranaki"}:
+            self._add_subdiv_tki_holidays()
+        elif self.subdiv == "Southland":
+            self._add_subdiv_stl_holidays()
+        elif self.subdiv == "Wellington":
+            self._add_subdiv_wgn_holidays()
+        elif self.subdiv in {"West Coast", "WTL", "Westland"}:
+            self._add_subdiv_wtc_holidays()
+
+    def _add_subdiv_holidays(self):
+        if self.subdiv in {"South Canterbury", "STC"}:
+            self._add_holiday(
+                "South Canterbury Anniversary Day",
+                _get_nth_weekday_of_month(4, MON, SEP, self._year),
+            )
```

Comment:
If "South Canterbury" and "STC" are the same we could create `def _add_subdiv_stc_holidays(self):` and add one more `elif` line above `elif self.subdiv in {"South Canterbury", "STC"}` invoking that method.

My point is `_add_subdiv_holidays` looks a bit too much for this.